reporter.go

```diff
@@ -216,12 +216,14 @@ func (r *remoteReporter) Report(span *Span) {
 
 // Close implements Close() method of Reporter by waiting for the queue to be drained.
 func (r *remoteReporter) Close() {
-	// First, cut off report requests still in-flight
-	close(r.closed)
-
-	// After that, drain the queue
+	// Cut off report requests still in-flight and drain the queue
+	// NB we keep r.queue open because it causes a race condition in Report() where since both
```

Comment:
why not close `r.queue` when exiting `processQueue()`?

Reply:
then we run into the same race condition in Report()